blocksync/worker.go

```diff
@@ -100,11 +107,15 @@ func (w *syncWorker) Sync() {
 		log.L().Warn("Error when get neighbor peers.", zap.Error(err))
 		return
 	}
-	intervals := w.buf.GetBlocksIntervalsToSync(w.targetHeight)
+	targetHeight := uint64(0)
+	if !forceSync {
+		targetHeight = w.targetHeight
+	}
+	intervals := w.buf.GetBlocksIntervalsToSync(targetHeight)
```

Comment:
nit: this seems not necessary?
GetBlocksIntervalsToSync() will make `confirmedHeight + b.bufferSize <= targetHeight <= confirmedHeight + b.intervalSize` anyway